test/credentials_test.dart

```diff
@@ -299,4 +299,143 @@ Future<void> main([List<String>? args]) async {
+    final app = App(configuration);
+    String username = "JWT_privatekey_validated_user@realm.io";
+    String userId = "62f394e9bcb9fee0c9aecb76";
+    var token =
+        "eyJraWQiOiIxIiwiYWxnIjoiUlMyNTYiLCJ0eXAiOiJKV1QifQ.eyJzdWIiOiI2MmYzOTRlOWJjYjlmZWUwYzlhZWNiNzYiLCJuYW1lIjp7ImZpcnN0TmFtZSI6IkpvaG4iLCJsYXN0TmFtZSI6IkRvZSJ9LCJlbWFpbCI6IkpXVF9wcml2YXRla2V5X3ZhbGlkYXRlZF91c2VyQHJlYWxtLmlvIiwiZ2VuZGVyIjoibWFsZSIsImJpcnRoRGF5IjoiMTk5OS0xMC0xMSIsIm1pbkFnZSI6IjEwIiwibWF4QWdlIjoiOTAiLCJjb21wYW55IjoiUmVhbG0iLCJpYXQiOjE2NjAxNDU2ODYsImV4cCI6NDgxMzc0NTY4NiwiYXVkIjoibW9uZ29kYi5jb20iLCJpc3MiOiJodHRwczovL3JlYWxtLmlvIn0.NAy60d4zpzRyJayO9qa6i7T3Yui4vrEJNK5FYhlQGAPCCKmPpBBrPZnOH2QwTsE1sW5jr9EsUPix6PLIauSY4nE-s4JrFb9Yu1QmhzYiXAzzyRK_yJOLmrOujqnWb57Z1KvZo5CsUafTgB5-mbs4t4-udIZEubEgr7sgH51rHK7F1r7EArwT3Fbx-EjPDTN1cWn4945Hku6wk0WgdXwVg6TEaNtT0RrEegw9t63sW1UvOYsgXpHfCePGH8VRX7yYYqu1xBnS1S1ZHNgGNZp3t8pu4lod6jHho0dPetAq9oMSmUP9H2uiKkwqFmWC_bVEjTxX4bGSbLGKZQRkiOn38w";
+    final credentials = Credentials.jwt(token);
+    final user = await app.logIn(credentials);
+
+    expect(user.state, UserState.loggedIn);
+    expect(user.identities[0].id, userId);
+    expect(user.provider, AuthProviderType.jwt);
+    expect(user.profile.email, username);
+    expect(user.profile.name, username);
+    expect(user.profile.gender, "male");
+    expect(user.profile.birthDay, "1999-10-11");
+    expect(user.profile.minAge, "10");
+    expect(user.profile.maxAge, "90");
+    expect(user.profile.firstName, "John");
+    expect(user.profile.lastName, "Doe");
+    expect(user.profile["company"], "Realm");
+  });
+
+  /// JWT Payload data
+  /// {
+  ///   "sub": "62f3840b4ac43f38a50b9e2b", //User.identities.id of the existing user realm-test@realm.io
+  ///   "name": {
+  ///     "firstName": "John",
+  ///     "lastName": "Doe"
+  ///   },
+  ///   "email": "jwt_user@#r@D@realm.io",
+  ///   "gender": "male",
+  ///   "birthDay": "1999-10-11",
+  ///   "minAge": "10",
+  ///   "maxAge": "90",
+  ///   "company": "Realm",
+  ///   "iat": 1660145055,
+  ///   "exp": 4813745055, //100 years after Aug 2022
+  ///   "aud": "mongodb.com",
+  ///   "iss": "https://realm.io"
+  /// }
+  baasTest('JWT - login with existing user and edit profile', (configuration) async {
+    final app = App(configuration);
+    final username = "jwt_user@#r@D@realm.io";
+    final authProvider = EmailPasswordAuthProvider(app);
+    // Always register jwt_user@#r@D@realm.io as a new user.
+    try {
+      await authProvider.registerUser(username, strongPassword);
+    } on RealmException catch (e) {
+      {
+        if (e.message.contains("name already in use")) {
+          // If the user exists, delete it and register a new one with the same name and empty profile
+          final user1 = await loginWithRetry(app, Credentials.emailPassword(username, strongPassword));
+          await app.deleteUser(user1);
+          await authProvider.registerUser(username, strongPassword);
+        }
+      }
```

Comment:
Why hardcode the username here? We should be able to use a random email and avoid that extra logic.

Reply:
Because the token is already generated and it must contain the same username inside.

Reply:
The logic gets complicated but it will allow us to run the same test many times.

Reply:
I tried to generate a new username for email/password credentials and then to link to JWT credentials. The email and username of the user are updated with "jwt_user@#r@D@realm.io" that is coming from the token. If I run the test again it creates a new user, but when I try to link it to the credentials with the same token it fails, because "jwt_user@#r@D@realm.io" already exists and it can not rename the new user to the same name. I still have to delete "jwt_user@#r@D@realm.io" before to run the test again.

Reply:
Yeah, the issue is that the token is already consumed to create a user. The next time the test runs, it tries to link the new user to an existing user which is forbidden.